src/views/TechnicalAssistance/RequestForm/Documents.tsx

```diff
@@ -1,29 +1,501 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
-import { useHistory } from 'react-router-dom';
+import {
+  Route,
+  Switch,
+  useHistory,
+  useParams,
+  useRouteMatch
+} from 'react-router-dom';
+import { CellProps, Column, useSortBy, useTable } from 'react-table';
+import { useMutation, useQuery } from '@apollo/client';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  Alert,
+  Button,
+  ErrorMessage,
+  Fieldset,
+  FileInput,
+  Form,
+  FormGroup,
+  Grid,
+  IconArrowBack,
+  Label,
+  Link,
+  Radio,
+  Table,
+  TextInput
+} from '@trussworks/react-uswds';
+import { clone } from 'lodash';
+import { DateTime } from 'luxon';
+
+import UswdsReactLink from 'components/LinkWrapper';
+import PageHeading from 'components/PageHeading';
+import CreateTrbRequestDocumentQuery from 'queries/CreateTrbRequestDocumentQuery';
+import DeleteTrbRequestDocumentQuery from 'queries/DeleteTrbRequestDocumentQuery';
+import GetTrbRequestDocumentsQuery from 'queries/GetTrbRequestDocumentsQuery';
+import {
+  CreateTrbRequestDocument,
+  CreateTrbRequestDocumentVariables
+} from 'queries/types/CreateTrbRequestDocument';
+import {
+  DeleteTrbRequestDocument,
+  DeleteTrbRequestDocumentVariables
+} from 'queries/types/DeleteTrbRequestDocument';
+import {
+  GetTrbRequestDocuments,
+  GetTrbRequestDocuments_trbRequest_documents as TrbRequestDocuments,
+  GetTrbRequestDocumentsVariables
+} from 'queries/types/GetTrbRequestDocuments';
+import {
+  TRBDocumentCommonType,
+  TRBRequestDocumentStatus
+} from 'types/graphql-global-types';
+import { getColumnSortStatus, getHeaderSortIcon } from 'utils/tableSort';
+import {
+  documentSchema,
+  TrbRequestInputDocument
+} from 'validations/trbRequestSchema';
+
+import Breadcrumbs from '../Breadcrumbs';
 
 import Pager from './Pager';
 import { FormStepComponentProps } from '.';
 
-function Documents({ request, stepUrl, taskListUrl }: FormStepComponentProps) {
+/**
+ * Documents is a component of both the table list of uploaded documents
+ * and the document upload form.
+ */
+function Documents({
+  request,
+  stepUrl,
+  taskListUrl,
+  setFormAlert
+}: FormStepComponentProps) {
   const { t } = useTranslation('technicalAssistance');
   const history = useHistory();
-  return (
-    <Pager
-      back={{
-        onClick: () => {
-          history.push(stepUrl.back);
+  const { url } = useRouteMatch();
+
+  const { view } = useParams<{
+    view?: string;
+  }>();
+
+  const { data, refetch /* , error, loading */ } = useQuery<
```

Comment:
I think it may be useful to render a loading state.  It works pretty quickly in local, but in env with lots of documents, this could render a blank table for a noticeable time.

I don't see any error Figma mock for fetching documents, so I guess it's fine for now to have the fallback be an empty array.  There may be a case where documents exist and it fails to fetch, which could be confusing to the user.

Reply:
Will add a loading spinner.

I was starting to assume a fetch error would be another one of the top level error types that appear in the header component.